src/Http/Http.Results/src/ObjectHttpResult.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http;
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+/// <summary>
+/// An <see cref="IResult"/> that on execution will write an object to the response.
+/// </summary>
+internal sealed class ObjectHttpResult : IResult
```

Comment:
Is this meant to be internal? If not, I don't see anything using it so it should be deleted?

Reply:
I will delete it in another PR, since there is a bunch of unit tests that might be needed to move.